### PR TITLE
Add --optimize/--no-optimize option for syncs

### PR DESCRIPTION
**Pull request by quba42**

### CODE REVIEW

**Review by mdellweg — APPROVED** [review]

File: tests/scripts/pulp_deb/test_deb_sync_publish.sh [hunk 1]
```diff
@@ -26,10 +26,18 @@ expect_succ pulp deb remote create \
 
 expect_succ pulp deb repository create \
   --name "${ENTITIES_NAME}_repo"
+  --remote "${ENTITIES_NAME}_remote"
 
 expect_succ pulp deb repository sync \
   --name "${ENTITIES_NAME}_repo" \
```
Comment (mdellweg) [inline tests/scripts/pulp_deb/test_deb_sync_publish.sh:32]:
```suggestion
  --name "${ENTITIES_NAME}_repo"
```

File: pulpcore/cli/deb/repository.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def repository(ctx: click.Context, pulp_ctx: PulpContext, repo_type: str) -> Non
         "since the last sync. This greately improves re-sync performance in such cases. Disable if "
         "the sync result does not match expectations."
     ),
-    needs_plugins=[PluginRequirement("deb", min="2.20.0")],
+    needs_plugins=[PluginRequirement("deb", min="2.20.0.dev")],
```
Comment (mdellweg) [inline pulpcore/cli/deb/repository.py:102]: This version spec is absolutely legit. We could clean it up but also leave it...

File: tests/scripts/pulp_deb/test_deb_sync_publish.sh [hunk 3]
```diff
@@ -25,11 +25,19 @@ expect_succ pulp deb remote create \
...
   --name "${ENTITIES_NAME}_repo"
 
 expect_succ pulp deb repository sync \
   --name "${ENTITIES_NAME}_repo" \
-  --remote "${ENTITIES_NAME}_remote"
+  --optimize
+
+expect_succ pulp deb repository sync \
+  --name "${ENTITIES_NAME}_repo" \
+  --no-optimize
```
Comment (mdellweg) [inline tests/scripts/pulp_deb/test_deb_sync_publish.sh:40]: You need to guard these lines with something like
`if pulp debug has-plugin --name deb --min-version 2.20; then`

Comment (quba42) [reply tests/scripts/pulp_deb/test_deb_sync_publish.sh:40]: Thanks for the hint.

File: pulpcore/cli/deb/repository.py (hunk 2 above)
Comment (quba42) [reply pulpcore/cli/deb/repository.py:108]: I squashed the DROP_ME commit, so this change is now part of the single commit.

File: tests/scripts/pulp_deb/test_deb_sync_publish.sh (hunk 3 above)
Comment (quba42) [reply tests/scripts/pulp_deb/test_deb_sync_publish.sh]: The extra syncs for the option are now conditional on the plugin version.

**Pull request merged**